Source/ACE/Network/NetworkSession.cs

```diff
@@ -12,110 +13,124 @@ namespace ACE.Network
 {
     public class NetworkSession
     {
+        private readonly Object lockObject = new Object();

```

Comment:
I strongly prefer lock objects to be "[objectBeingLocked]Lock" and directly adjacent to it.  This appears to be a functional lock for both HandlePacket and EnqueueSend, and I dont' see the necessity of those sharing a lock.  In this case, I would have named it "currentBundleLock" or "currentBundleMutex" - either are good.